web/src/components/ak-multi-select.ts

```diff
@@ -0,0 +1,150 @@
+import PFBase from "@patternfly/patternfly/patternfly-base.css";
+
+type Pair = [string, string];
+
+const selectStyles = css`
+    select[multiple] {
+        min-height: 15rem;
+    }
+`;
+
+/**
+ * Horizontal layout control with a multi-select.
+ *
+ * @part select - The select itself, to override the height specified above.
+ */
+@customElement("ak-multi-select")
+export class AkMultiSelect extends AKElement {
+    constructor() {
+        super();
+        this.dataset.akControl = "true";
+    }
+
+    static get styles() {
+        return [PFBase, PFForm, PFFormControl, selectStyles];
+    }
+
+    /**
+     * The [name] attribute, which is also distributed to the layout manager and the input control.
+     */
+    @property({ type: String })
+    name!: string;
+
+    /**
+     * The text label to display on the control
+     */
+    @property({ type: String })
+    label = "";
+
+    /**
+     * The values to be displayed in the select. The format is [Value, Label], where the label is
+     * what will be displayed.
+     */
+    @property({ attribute: false })
+    options: Pair[] = [];
+
+    /**
+     * If true, at least one object must be selected
+     */
+    @property({ type: Boolean })
+    required = false;
+
+    /**
+     * Supporting a simple help string
+     */
+    @property({ type: String })
+    help = "";
+
+    /**
+     * For more complex help instructions, provide a template result.
+     */
+    @property({ type: Object })
+    bighelp!: TemplateResult | TemplateResult[];
+
+    /**
+     * An array of strings representing the objects currently selected.
+     */
+    @property({ type: Array })
+    values: string[] = [];
+
+    /**
+     * Helper accessor for older code
+     */
+    get value() {
+        return this.values;
+    }
+
+    /**
+     * One of two criteria (the other being the data-ak-control flag) that specifies this as a
+     * control that produces values of specific interest to our REST API. This is our modern
+     * accessor name.
+     */
+    json() {
+        return this.values;
+    }
+
+    renderHelp() {
+        return [
+            this.help ? html`<p class="pf-c-form__helper-text">${this.help}</p>` : nothing,
+            this.bighelp ? this.bighelp : nothing,
+        ];
+    }
+
+    handleChange(ev: Event) {
+        if (ev.type === "change") {
+            this.values = Array.from(this.selectRef.value!.querySelectorAll("option"))
+                .filter((option) => option.selected)
+                .map((option) => option.value);
+            this.dispatchEvent(
+                new CustomEvent("ak-select", {
+                    detail: this.values,
+                    composed: true,
+                    bubbles: true,
+                }),
+            );
+        }
+    }
+
+    selectRef: Ref<HTMLSelectElement> = createRef();
+
+    render() {
+        return html` <div class="pf-c-form">
```

Comment:
This `div` that brings in all of the PF-Base and PF-Grid definitions turns out to be what has been missing from the formula for building proper, closed controls using Patternfly.